### PR TITLE
[PM-7740] Create notifications settings component (navigational changes)

### PR DESCRIPTION
**Pull request by djsmith85**
## Type of change

```
- [ ] Bug fix
- [X] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Split out the appearance-settings related controls into a separate page according to the UI re-design (navigational changes)

This only contains navigational changes and makes it easier to enable code-ownership
settings.component is under tools-ownership
and
notifications.component is under autofill-ownership

The visual changes planned for the UI refresh will be done, once all needed base components from the Component Library are available.

## Code changes

- **Create notifcation settings component -** https://github.com/bitwarden/clients/commit/9c441302ca0e42cf800816d5020db5d561737158
- **Move excluded-domains component over to be owned by team-autofill -** https://github.com/bitwarden/clients/commit/2f8ccca5bb820a4711b4f2be8992234c2f9502ac
- **Add notifications entry to settings screen -** https://github.com/bitwarden/clients/commit/7fd29116cb4593e12cd957ddddd51f78d934d363
- **Move excluded domains from settings to notifications settings screen -** https://github.com/bitwarden/clients/commit/d6f7f477499ca4bc5b0b4002116870a94a1aa6ef

### Before
before_settings
before_options

### After
after_settings
after_notifications

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

**Review by jprusik — DISMISSED** [review]
LGTM

File: apps/browser/src/popup/app-routing.module.ts [hunk 1]
```diff
@@ -251,6 +252,12 @@ const routes: Routes = [
     canActivate: [AuthGuard],
     data: { state: "account-security" },
   },
+  {
+    path: "notifications",
```
Comment (audrey-jensen) [inline apps/browser/src/popup/app-routing.module.ts:256]: 🤔 I wonder whether this should include an indication that it's tailoring the notification settings? I'd expect  `/notifications` to display a notification log.

**Review by audrey-jensen — DISMISSED** [review]
🤔 non-blocking concern

**Review by audrey-jensen — DISMISSED** [review]

**Review by audrey-jensen — DISMISSED** [review]

**Review by audrey-jensen — APPROVED** [review]

**Comment by jprusik** [comment]
BIT is happy:
https://github.com/bitwarden/browser-interactions-testing/actions/runs/9006734834
https://github.com/bitwarden/browser-interactions-testing/actions/runs/9006736542

and the new changes LGTM 👍

**Review by jprusik — APPROVED** [review]
BIT is happy:
https://github.com/bitwarden/browser-interactions-testing/actions/runs/9006734834
https://github.com/bitwarden/browser-interactions-testing/actions/runs/9006736542

and the new changes LGTM 👍

**Pull request merged**